pkg/generic/thrift/write.go

```diff
@@ -397,13 +429,33 @@ func writeList(ctx context.Context, val interface{}, out thrift.TProtocol, t *de
 	if length == 0 {
 		return out.WriteListEnd()
 	}
-	writer, err := nextWriter(l[0], t.Elem, opt)
-	if err != nil {
-		return err
+	var (
+		writer, defaultWriter writer
+		zeroValue             interface{}
+		err                   error
+	)
+	for _, elem := range l {
+		if elem != nil {
+			if writer, err = nextWriter(elem, t.Elem, opt); err != nil {
+				return err
+			}
+			break
+		}
 	}
 	for _, elem := range l {
-		if err := writer(ctx, elem, out, t.Elem, opt); err != nil {
-			return err
+		if elem == nil {
+			if defaultWriter == nil {
+				if zeroValue, defaultWriter, err = getDefaultValueAndWriter(t.Elem, opt); err != nil {
+					return err
+				}
+			}
+			if err := defaultWriter(ctx, zeroValue, out, t.Elem, opt); err != nil {
+				return err
+			}
+		} else {
+			if err := writer(ctx, elem, out, t.Elem, opt); err != nil {
+				return err
+			}
```

Comment:
It looks a little bit confusing. It might be more readable and efficient that we get `writer` when we need one, like below:
```suggestion
                        if writer == nil {
                            if writer, err = nextWriter(elem, t.Elem, opt); err != nil {
				return err
			    }
			}
			if err := writer(ctx, elem, out, t.Elem, opt); err != nil {
				return err
			}
```
then we don't need the above [loop](https://github.com/cloudwego/kitex/pull/718/files#diff-546fbeb16d59473ad5c0dd7a46fd2e45bd11b65498fb8c0ac80408aeb35b787bR437-R444) to get writer